addon/hint/show-hint.js

```diff
@@ -42,8 +42,9 @@
     this.widget = null;
     this.debounce = 0;
     this.tick = 0;
-    this.startPos = this.cm.getCursor();
+    this.startPos = this.cm.getCursor("start");
     this.startLen = this.cm.getLine(this.startPos.line).length;
+    if (options && options.somethingSelected == false ) this.startLen = this.startLen - this.cm.getSelection().length;
```

Comment:
> Ah, somethingSelected: false is how you indicate that completing with selected text is allowed?

Yes that it is! Perhaps my code is bad. My need is to open completion with selected text and adjust startLen according the selection.